noir_stdlib/src/ec/swcurve.nr

```diff
@@ -344,11 +344,20 @@ mod curvegroup {
 
         // Scalar multiplication (p + ... + p n times)
         fn mul(self, n: Field, p: Point) -> Point {
-            let n_as_bits = crate::ec::to_bits(n); // N_BITS-bit representation
+            let N_BITS = crate::field::modulus_num_bits();
 
-            self.bit_mul(n_as_bits, p)
```

Comment:
Why don't you use a function for bit_mul?

Reply:
Yes, that would be ideal, but I get a panic if I change to this and do `nargo compile`:
```rust
// Scalar multiplication with scalar represented by a bit array (little-endian convention).
// If k is the natural number represented by `bits`, then this computes p + ... + p k times.
fn bit_mul<N>(self, bits: [u1; N], p: Point) -> Point {
    let mut out = Point::zero();
    let n = bits.len();
    
    for i in 0..n {
        out = self.add(
            self.add(out, out),
            if(bits[n - i - 1] == 0) {Point::zero()} else {p});
    }

    out
}

// Scalar multiplication (p + ... + p n times)
fn mul(self, n: Field, p: Point) -> Point {
    let N_BITS = crate::field::modulus_num_bits();
    let n_as_bits = n.to_le_bits(N_BITS as u32);

    self.bit_mul(n_as_bits, p)
}
```

Reply:
Could you create an issue with the error you get?

Until the issue is resolved, you can try this which seems to work (and you would reference the issue):
```
            //temporary workaround until issue XXX is solved
            let mut n_as_bits :[u1;254] = [0;254];
            let tmp = n.to_le_bits(N_BITS as u32);
            for i in 0..254 {
               n_as_bits[i]= tmp[i]; 
            }
           self.bit_mul(n_as_bits, p)
```

Reply:
Okay, I did this. Though it adds a field-dependent constant that I don't like
Issue: #1354